### PR TITLE
Implements fluent API - addDirectory,addBundles,addDirectories

### PR DESCRIPTION
**Pull request by goochjj**
This also allows TestBox to look in multiple specs folders.
For instance:
  tests/specs/
  modules_app/api-v1/tests/specs/

### CODE REVIEW

File: system/TestBox.cfc [hunk 1]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
+	* @directory A directory to test which can be a simple mapping path or a struct with the following options: [ mapping = the path to the directory using dot notation (myapp.testing.specs), recurse = boolean, filter = closure that receives the path of the CFC found, it must return true to process or false to continue process ]
+	*/
+	any function addDirectory(required any directory, boolean recurse = true) {
 		// inflate directory?
-		if( isSimpleValue( arguments.directory ) ){ arguments.directory = { mapping=arguments.directory, recurse=true }; }
+		if( isSimpleValue( arguments.directory ) ) { arguments.directory = { mapping=arguments.directory, recurse=arguments.recurse }; }
 		// directory passed?
 		if( !structIsEmpty( arguments.directory ) ){
-			arguments.bundles = getSpecPaths( arguments.directory );
+			variables.bundles = ArrayMerge(variables.bundles, getSpecPaths( arguments.directory ));
```
Comment (lmajano) [inline system/TestBox.cfc]: `arrayMerge` is a lucee only function, we have to support ACF10+

File: system/TestBox.cfc [hunk 2]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
-		inflateLabels( arguments.labels );
-		// inflate bundles to array
-		inflateBundles( arguments.bundles );
+	/**
+	* Constructor
+	* @directories A set of directories to test which can be a list of simple mapping paths or an array of structs with the following options: [ mapping = the path to the directory using dot notation (myapp.testing.specs), recurse = boolean, filter = closure that receives the path of the CFC found, it must return true to process or false to continue process ]
+	*/
+	any function addDirectories(required any directories, boolean recurse = true) {
+		// inflate directory?
+		if( isSimpleValue( arguments.directories ) ) { arguments.directories = ListToArray(arguments.directories); }
```
Comment (lmajano) [inline system/TestBox.cfc]: Spacing please :)

`listToArray( arguments.directories )`

File: system/TestBox.cfc [hunk 3]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
+		// directory passed?
+		if( !arrayIsEmpty( arguments.directories ) ){
```
Comment (lmajano) [inline system/TestBox.cfc]: You can actually eliminate this check, the for loop will just skip if the target is empty

File: system/TestBox.cfc [hunk 4]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
+			for(var dir in arguments.directories) {
```
Comment (lmajano) [inline system/TestBox.cfc]: spacing please:
`for( var in arguments.directories ){`

File: system/TestBox.cfc [hunk 5]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
+				addDirectory(dir, arguments.recurse);
```
Comment (lmajano) [inline system/TestBox.cfc]: Spacing: `addDirectory( dir, arguments.recurse );`

File: system/TestBox.cfc [hunk 6]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
+		}
+		return this;
+	}
 
+	/**
+	* Constructor
+	* @directory A directory to test which can be a simple mapping path or a struct with the following options: [ mapping = the path to the directory using dot notation (myapp.testing.specs), recurse = boolean, filter = closure that receives the path of the CFC found, it must return true to process or false to continue process ]
+	*/
+	any function addBundles(required any bundles) {
+		if( isSimpleValue( arguments.bundles ) ) { arguments.bundles = listToArray(arguments.bundles); }
```
Comment (lmajano) [inline system/TestBox.cfc]: Spacing: `listToArray( arguments.bundles )`

File: system/TestBox.cfc [hunk 7]
```diff
@@ -50,19 +52,60 @@ component accessors="true"{
...
+		if( !arrayIsEmpty( arguments.bundles ) ) {
+			variables.bundles = ArrayMerge(variables.bundles, arguments.bundles);
```
Comment (lmajano) [inline system/TestBox.cfc]: We can't use `arrayMerge` to support acf10+

**Comment by goochjj** [comment]
Updated the PR as requested!

File: system/TestBox.cfc (hunk 1 above)
Comment (lmajano) [inline system/TestBox.cfc]: @goochjj This still appears as `arrayMerge`

**Comment by goochjj** [comment]
I swear I did those changes once.  But when I went to pull from the repo it only had the original commit I did. (I even remember the review bits doing the "show outdated" thing they're doing now)  I wonder if I managed to clobber it again later somehow. Anyway.

ArrayMerge is gone and the whitespace should match your preferences.

**Pull request merged**